### PR TITLE
fix driver for old ocl kernel

### PR DESCRIPTION
**Pull request by bghimireamd**
Initialize tensors in batch norm driver for OCL and CK kernel correctly.

### CODE REVIEW

File: driver/bn_driver.hpp [hunk 1]
```diff
@@ -1470,14 +1481,14 @@ int BatchNormDriver<Tgpu, Tref, Tmix>::VerifyBackward()
...
-            diff   = fabs(Tgpu(fabs(out_ref.data[i]) - fabs(out_bwd.GetVector()[i])));
+            diff   = fabs(TInput(fabs(out_ref.data[i]) - fabs(out_bwd.GetVector()[i])));
             maxval = maxval < diff ? diff : maxval;
             if(!std::isfinite(diff) || diff > tolerance)
             {
                 std::cout << "out_ref[" << i << "]: " << out_ref.data[i];
                 std::cout << "\tout_bwd.GetVector()[" << i << "]: " << out_bwd.GetVector()[i];
                 std::cout << "\tdiff[" << i
-                          << "]: " << Tgpu(fabs(out_ref.data[i]) - fabs(out_bwd.GetVector()[i]));
+                          << "]: " << TInput(fabs(out_ref.data[i]) - fabs(out_bwd.GetVector()[i]));
```
Comment (BrianHarrisonAMD) [inline driver/bn_driver.hpp]: I think some of these types might need to be changed now.
This is using out_bwd, and I think that needs to be TOut now right?

File: driver/bn_driver.hpp [hunk 2]
```diff
@@ -183,80 +188,80 @@ int BatchNormDriver<Tgpu, Tref, Tmix>::ParseCmdLineArgs(int argc, char* argv[])
...
+        out_bwd.AllocOnHost(tensor<TCKOCLmix>{bn_layout, in_len});
 
-        bnScale.AllocOnHost(tensor<Tgpu>{bn_layout, derivedBnDesc.GetLengths()});
-        dy.AllocOnHost(tensor<Tmix>{bn_layout, in_len});
+        bnScale.AllocOnHost(tensor<TScaleBias>{bn_layout, derivedBnDesc.GetLengths()});
+        dy.AllocOnHost(tensor<TCKOCLmix>{bn_layout, in_len});
 
         auto gen_var_bwd = [](auto...) {
-            return static_cast<Tmix>(1e-2 * (prng::gen_0_to_B(100) + 1));
+            return static_cast<TCKOCLmix>(1e-2 * (prng::gen_0_to_B(100) + 1));
```
Comment (BrianHarrisonAMD) [inline driver/bn_driver.hpp]: These are using the wrong type after rename I think.

File: driver/bn_driver.hpp [hunk 3]
```diff
@@ -183,80 +188,80 @@ int BatchNormDriver<Tgpu, Tref, Tmix>::ParseCmdLineArgs(int argc, char* argv[])
     return miopenStatusSuccess;
 }
 
-template <typename Tgpu, typename Tref, typename Tmix>
-int BatchNormDriver<Tgpu, Tref, Tmix>::GetandSetData()
+template <typename TInput, typename Tref, typename TAcc, typename TScaleBias, typename TCKOCLmix>
+int BatchNormDriver<TInput, Tref, TAcc, TScaleBias, TCKOCLmix>::GetandSetData()
```
Comment (BrianHarrisonAMD) [inline driver/bn_driver.hpp]: I think TCKOCLmix was renamed to TOut, and there is quite a few places that are still using the old name.

File: driver/bn_driver.hpp (hunk 1 above)
Comment (bghimireamd) [reply driver/bn_driver.hpp]: yes! for out_bwd

**Review by CAHEK7 — APPROVED** [review]

**Review by BrianHarrisonAMD — APPROVED** [review]
LGTM!

**Review by junliume — APPROVED** [review]

**Pull request merged**